pkg/test/expectations/expectations.go

```diff
@@ -137,3 +145,37 @@ func ExpectReconcileSucceeded(reconciler reconcile.Reconciler, key client.Object
 	_, err := reconciler.Reconcile(context.Background(), reconcile.Request{NamespacedName: key})
 	Expect(err).ToNot(HaveOccurred())
 }
+
+func ExpectPodsEvictingSucceeded(c client.Client, pods ...*v1.Pod) {
+	for _, pod := range pods {
+		Eventually(func() bool {
+			return ExpectPodExists(c, pod.Name, pod.Namespace).GetDeletionTimestamp().IsZero()
+		}, ReconcilerPropagationTime, RequestInterval).Should(BeFalse(), func() string {
+			return fmt.Sprintf("expected %s/%s to be evicting, but it isn't", pod.Namespace, pod.Name)
+		})
+	}
+}
+
+// ExpectPodsEvictingFailed checks if an eviction fails by checking if it has been requeued
+func ExpectPodsEvictingFailed(c client.Client, queue workqueue.RateLimitingInterface, pods ...*v1.Pod) {
+	for _, pod := range pods {
+		Eventually(func() bool {
+			return ExpectPodExists(c, pod.Name, pod.Namespace).GetDeletionTimestamp().IsZero() &&
+				queue.NumRequeues(types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}) > 0
+		}, ReconcilerPropagationTime, RequestInterval).Should(BeTrue(), func() string {
+			return fmt.Sprintf("expected %s/%s to not be evicting, but it is", pod.Namespace, pod.Name)
+		})
+	}
+}
+
+func ExpectPodsEnqueuedForEviction(c client.Client, enqueued set.Set, pods ...*v1.Pod) {
+	for _, pod := range pods {
+		Expect(enqueued.Contains(types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace})).To(BeTrue())
+	}
+}
+
+func ExpectPodsNotEnqueuedForEviction(c client.Client, enqueued set.Set, pods ...*v1.Pod) {
```

Comment:
IMO these methods are way to specific to the eviction logic to live in this package. I don't want to dilute the cohesion of this package, so can we put them at the bottom of suite_test.go in the eviction package?